### PR TITLE
feat(types): add missing typescript definitions for reporters

### LINKED ISSUE #92: missing typescript definitions for reporters
### Version

[v2.11.3](https://github.com/consola/releases/tag/v2.11.3)

### Reproduction link

[https://example.com](https://example.com)

### Steps to reproduce

import { Consola, FancyReporter } from 'consola';
 will produce an error as there is no exported member `FancyReporter` 

### What is expected ?

to have proper detentions for the entire library

### What is actually happening?

typescript error

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/consola/issues/c70">#c70</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by regevbr**
Fix #92

### CODE REVIEW

**Comment by regevbr** [comment]
@Atinux @pi0 can you please CR this?

**Comment by regevbr** [comment]
There are 2 drawback here

In order to provide proper typings for the winston reporter, we must import it. But if someone doesn't use it, it will produce a compilation error. This error can be prevented by setting in `tsconfig.json`
```json
"skipLibCheck": true,
```
I usually use that flag any ways, but I guess some people will discourage it. I don't mind making the logger type there to be `any` if you prefer.

I added typings for the log levels. The silent level is set to be Infinity in the code. But typescript doesn't provide a valid way to use it a type literal - microsoft/TypeScript/issues/32277. To overcome it I set it to be a random really high number.
Since the value of infinity is not really used by the code, and only used to denote that if someone inputs a really high level, it will effectively not log anything, I think my solution will work, but I believe it will be better to set a reasonable high number (such as 9999) to be marked as the silent log level. 

What do you think?

File: types/consola.d.ts [hunk 1]
```diff
@@ -1,10 +1,35 @@
...
+export type FatalLogLevel = 0;
+export type ErrorLogLevel = 0;
+export type WarnLogLevel = 1;
+export type LogLogLevel = 2;
+export type InfoLogLevel = 3;
+export type SuccessLogLevel = 3;
+export type DebugLogLevel = 4;
+export type TraceLogLevel = 5;
+export type SilentLogLevel = Infinity;
+export type LogLevel =
```
Comment (pi0) [inline types/consola.d.ts:14]: Can we instead use an enum?

**Comment by pi0** [comment]
Hi @regevbr thanks for the PR. I think based on the amount of new types it is better to rewrite lib to typescript to auto-generate them.

**Comment by regevbr** [comment]
Hi rewriting will take a long time and I'm no available for it right now as I'm working on other projects. Do tou think we can use those typings for now?

File: types/consola.d.ts (hunk 1 above)
Comment (regevbr) [reply types/consola.d.ts:14]: Enums won't work well on d.ts alone as you must expose a real object to represent them. I will see if I can add it to the js and then expose it as an Enum here

**Comment by pi0** [comment]
Of course, we can do it until rewrite. I'm just worried about winston and log level types. Can we, for now, omit Winston and use an enum (via js) for levels?

**Comment by regevbr** [comment]
Great!
Sure already implement the enum and I will drop winston

File: types/consola.d.ts (hunk 1 above)
Comment (regevbr) [reply types/consola.d.ts]: Fixed it, can you please CR it again?

**Comment by regevbr** [comment]
Thanks @pi0! when can we expect a new release with the changes?

**Comment by pi0** [comment]
@regevbr sorry for the delay. will be released in a few minutes. Thanks again for PR :heart: :)

**Comment by regevbr** [comment]
Sure thing! thanks :-)

**Comment by Atinux** [comment]
It's out: https://github.com/nuxt-contrib/consola/releases/tag/v2.12.0

Thank you @regevbr :)

**Comment by regevbr** [comment]
Thanks @Atinux

**Pull request merged**